src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -288,18 +288,53 @@ private async Task<TimeSpan> DownloadFullDatabaseAsync(FileInfo databaseFileInfo
 
             // Write the file out to disk so we'll have it the next time we launch VS.  Do this
             // after we set the in-memory instance so we at least have something to search while
-            // we're waiting to write.
-            await WriteDatabaseFileAsync(databaseFileInfo, bytes, cancellationToken).ConfigureAwait(false);
+            // we're waiting to write. It's ok if either of these writes don't succeed. If the txt
+            // file fails to persist, a subsequent VS session will redownload the index and again try
+            // to persist. If the binary file fails to persist, a subsequent VS session will just load
+            // the index from the txt file and again try to persist the binary file.
+            await WriteDatabaseTextFileAsync(databaseFileInfo, bytes, cancellationToken).ConfigureAwait(false);
+            await WriteDatabaseBinaryFileAsync(database, databaseFileInfo, cancellationToken).ConfigureAwait(false);
 
             var delay = _service._delayService.UpdateSucceededDelay;
             LogInfo($"Processing full database element completed. Update again in {delay}");
             return (succeeded: true, delay);
         }
 
-        private async Task WriteDatabaseFileAsync(FileInfo databaseFileInfo, byte[] bytes, CancellationToken cancellationToken)
+        private async Task WriteDatabaseTextFileAsync(FileInfo databaseFileInfo, byte[] bytes, CancellationToken cancellationToken)
         {
             LogInfo("Writing database file");
 
+            await WriteDatabaseFileAsync(databaseFileInfo, new ArraySegment<byte>(bytes), cancellationToken).ConfigureAwait(false);
+
+            LogInfo("Writing database file completed");
+        }
+
+        private async Task WriteDatabaseBinaryFileAsync(AddReferenceDatabase database, FileInfo databaseFileInfo, CancellationToken cancellationToken)
+        {
+            using var memoryStream = new MemoryStream();
+            using var writer = new BinaryWriter(memoryStream);
+
+            LogInfo("Writing database binary file");
+
+            database.WriteBinary(writer);
+            writer.Flush();
+
+            // Obtain the underlying array from the memory stream. If for some reason this isn't available,
+            // fall back to reading the stream into a new byte array.
+            if (!memoryStream.TryGetBuffer(out var arraySegmentBuffer))
```

Comment:
outside of this PR.  but i think we could benefit from thsi pattern in a few more places.  like in the sqlite system.  def takea  look.

Reply:
I didn't find anything with my quick search, was there a particular class you had in mind?

Reply:
i was thinking about Sqlite Accessor.WriteStream  which which works with streams and does some funky stuff with SQLitePersistentStorage.GetBytes to try to an array to write to to then write to the DB.